### DISCUSSION TITLE
头文件中的库函数如何处理

### DISCUSSION THREAD

**Discussion by hodanna7** [arcsysu/SYsU-lang#44] [Q&A] [answered]
头文件中的库函数似乎与自己创建的函数不同，在IR中没有对应的指令，应该怎么处理比较好？需要处理些什么？

**Comment by Yorkking** [comment, marked as answer]
那些函数就当成外部导入的，对应的 IR 类似这样的：
```C
declare void @_sysy_putch(i32)
```
对应的 LLVM 的 API 其实和本地函数一样，只不过后面不需要再对一来生成的 Function 再创建基本块插入函数内部代码就好了。

----
520 不过节？😋 520快乐！

**Reply by hodanna7** [reply]
呜呜呜谢谢助教QwQ
